src/main/java/ladder/domain/Ladder.java

```diff
@@ -0,0 +1,103 @@
+package ladder.domain;
+
+import ladder.domain.exception.ContinuousTrueStickSameHeightException;
+import ladder.domain.exception.DifferentLineSizeException;
+import ladder.domain.exception.LinesLessThanMinimumSizeException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Ladder {
+
+    private static final int DISTINCT_LINE_HEIGHT_COUNT = 1;
+    private static final int MINIMUM_LINES_SIZE = 1;
+
+    private final List<Line> lines;
+
+    public Ladder(final List<Line> lines) {
+        List<Line> nullSafeLines = Optional.ofNullable(lines)
+                .orElse(new ArrayList<>());
+        validate(nullSafeLines);
+        this.lines = nullSafeLines;
+    }
+
+    public Ladder(LinesGenerable linesGenerable) {
+        this(linesGenerable.generate());
+    }
+
+    private void validate(List<Line> lines) {
+        validateMinimumLinesSize(lines);
+        validateLinesAllSameHeight(lines);
+        validateConsecutiveSticks(lines);
+    }
+
+    private void validateMinimumLinesSize(List<Line> lines) {
+        if (lines.size() < MINIMUM_LINES_SIZE) {
+            throw LinesLessThanMinimumSizeException.from(MINIMUM_LINES_SIZE);
+        }
+    }
+
+    private void validateLinesAllSameHeight(List<Line> lines) {
+        long distinctLineHeightCount = lines.stream()
+                .map(Line::findHeight)
+                .distinct()
+                .count();
+        if (distinctLineHeightCount != DISTINCT_LINE_HEIGHT_COUNT) {
+            throw DifferentLineSizeException.getInstance();
+        }
+    }
+
+    private void validateConsecutiveSticks(List<Line> lines) {
+        if (lines.size() <= 1) {
+            return;
+        }
+        IntStream.range(0, lines.size() - 1)
+                .forEach(index -> checkConsecutiveLines(index, lines));
+    }
+
+    private void checkConsecutiveLines(int index, List<Line> lines) {
+        Line line = lines.get(index);
+        Line nextLine = lines.get(index + 1);
+        List<Integer> indexes = line.findIndexesTrueStickOfSameIndex(nextLine);
+        if (indexes.size() > 0) {
+            throw ContinuousTrueStickSameHeightException.getInstance();
+        }
+    }
+
+    public List<Stick> findSticksOf(LadderHeight height) {
+        return lines.stream()
+                .map(line -> line.findStickOf(height))
+                .collect(Collectors.toList());
+    }
+
+    public LadderHeight findHeight() {
+        return lines.stream()
+                .findFirst()
+                .map(Line::findHeight)
+                .orElse(new LadderHeight(1));
```

Comment:
사다리의 최소 높이가 1이라면, 생성자에 `1`이라는 숫자 리터럴을 전달하지 않고도
기본 생성자로 `LadderHeight` 인스턴스를 생성할 때 높이 값을 1로 부여해보는 것도 좋을 것 같아요.
아니면 상수인 `LadderHeight.MINIMUM`을 전달해볼 수도 있겠네요.
```suggestion
                .orElse(new LadderHeight());
```
```suggestion
                .orElse(new LadderHeight(LadderHeight.MINIMUM));
```

Reply:
네 확실히 1이라는 숫자를 상수로 표현하면 의미전달이 더 쉽게 되는 것 같아요:)